### PR TITLE
BP-3045-Implement-SonarQube-to-analyse-code

### PR DESCRIPTION
**Pull request by Rinor12010**

### CODE REVIEW

File: buckaroo3.php [hunk 1]
```diff
@@ -586,16 +557,13 @@ public function isPaymentModeActive($method)
...
-        if ($isLive === 0) {
-            return isset($configArray['mode']) && $configArray['mode'] === 'test';
-        } elseif ($isLive === 1) {
-            return isset($configArray['mode']) && $configArray['mode'] === 'live';
+        if (!empty($configArray)) {
+            if ($isLive === 0) {
+                return isset($configArray['mode']) && $configArray['mode'] === 'test';
+            } elseif ($isLive === 1) {
+                return isset($configArray['mode']) && $configArray['mode'] === 'live';
+            }
```
Comment (harli91) [inline buckaroo3.php:565]: is somewhat difficult to read

File: controllers/front/ajax.php [hunk 2]
```diff
@@ -33,90 +33,93 @@ public function postProcess()
...
-                if (!$paymentFee) {
-                    $presentedCart = $this->cart_presenter->present($cart);
-                    $this->context->smarty->assign([
-                        'configuration' => $this->getTemplateVarConfiguration(),
-                        'cart' => $presentedCart,
-                        'display_transaction_updated_info' => Tools::getIsset('updatedTransaction'),
-                    ]);
+                $this->getTotalCartPrice($currency,$locale);
+                break;
+            default:
```
Comment (harli91) [inline controllers/front/ajax.php:38]: why do you need this swich ????????

File: controllers/front/ajax.php [hunk 3]
```diff
@@ -33,90 +33,93 @@ public function postProcess()
...
+        }
+    }
+    private function getTotalCartPrice($currency,$locale){
```
Comment (harli91) [inline controllers/front/ajax.php:41]: this method is somewhat big, maybe split it into smaller methods

File: controllers/front/ajax.php [hunk 4]
```diff
@@ -33,90 +33,93 @@ public function postProcess()
...
+                'amount' => $total_including_tax,
+                'value' => $locale->formatPrice($total_including_tax, $currency->iso_code),
+            ],
+            'total_excluding_tax' => [
+                'type' => 'total',
+                'label' => $this->translator->trans('Total (tax excl.)', [], 'Shop.Theme.Checkout'),
+                'amount' => $total_excluding_tax,
+                'value' => $locale->formatPrice($total_excluding_tax, $currency->iso_code),
+            ],
+        ];
```
Comment (harli91) [inline controllers/front/ajax.php:108]: from what it looks to me we somehow override some prestashop variables, I am not sure if we want to do this, what happens if someone else decides to do the same?

**Review by vegimcarkaxhija — APPROVED** [review]

**Pull request merged**